src/pipe.js

```diff
@@ -1,34 +1,31 @@
-var compose = require('./compose');
-var reverse = require('./reverse');
+var _pipe = require('./internal/_pipe');
+var curryN = require('./curryN');
+var reduce = require('./reduce');
+var tail = require('./tail');
 
 
 /**
- * Creates a new function that runs each of the functions supplied as parameters in turn,
- * passing the return value of each function invocation to the next function invocation,
- * beginning with whatever arguments were passed to the initial invocation.
- *
- * `pipe` is the mirror version of `compose`. `pipe` is left-associative, which means that
- * each of the functions provided is executed in order from left to right.
+ * Performs left-to-right function composition. The leftmost function may have
+ * any arity; the remaining functions must be unary.
  *
  * In some libraries this function is named `sequence`.
+ *
  * @func
  * @memberOf R
  * @category Function
- * @sig ((a... -> b), (b -> c), ..., (x -> y), (y -> z)) -> (a... -> z)
- * @param {...Function} functions A variable number of functions.
- * @return {Function} A new function which represents the result of calling each of the
- *         input `functions`, passing the result of each function call to the next, from
- *         left to right.
+ * @sig ((a -> b), (b -> c), ..., (y -> z)) -> (a -> b -> c -> ... -> y -> z)
```

Comment:
Yes, that looks right.  My question, I guess is whether after your changes, if that is _still_ the correct sig?  This PR has gone through some radical changes.  The original one radically altered the API.  Are we now entirely back to this API; is it now, at least as far as `compose` / `pipe` a matter of implementation?  Or are there substantial API differences too?